trl/trainer/sft_trainer.py

```diff
@@ -228,6 +228,18 @@ def __init__(
         if processing_class is None:
             processing_class = AutoTokenizer.from_pretrained(model_id)
 
+        if args.eos_token is not None:
+            eos_token = args.eos_token
+            eos_token_id = processing_class.convert_tokens_to_ids(eos_token)
+            if eos_token_id is None:
+                raise ValueError(
+                    f"The specified `eos_token` ('{eos_token}') is not found in the vocabulary of the given "
+                    f"`processing_class` ({processing_class.__class__.__name__}). Ensure that the `eos_token` exists "
+                    "in the vocabulary before using it as an EOS token."
+                )
+            processing_class.eos_token = eos_token
+            processing_class.eos_token_id = eos_token_id
```

Comment:
I think only one of these is necessary. The other is automatically set. I can't remember which one.
But it's not important, we can keep it like this.

Reply:
Both work actually:

```python
>>> from transformers import AutoTokenizer
>>> tok = AutoTokenizer.from_pretrained("Qwen/Qwen2.5-0.5B")
>>> tok.eos_token, tok.eos_token_id
('<|endoftext|>', 151643)
>>> tok.eos_token = "eos"
>>> tok.eos_token, tok.eos_token_id
('eos', 84399)
>>> tok.eos_token_id = 84400
>>> tok.eos_token, tok.eos_token_id
('Ġwidening', 84400)
```

Reply:
```suggestion
            processing_class.eos_token_id = eos_token_id
```